coverage/coverage.mbt

```diff
@@ -0,0 +1,100 @@
+pub struct CoverageCounter {
+  priv counter : Array[Int]
+}
+
+fn CoverageCounter::new(size : Int) -> CoverageCounter {
+  { counter: Array::make(size, 0) }
+}
+
+fn CoverageCounter::incr(self : CoverageCounter, idx : Int) -> Unit {
+  let counter = self.counter[idx];
+  if counter < 0x7fffffff { // prevent overflow
+    self.counter[idx] = counter + 1
+  }
+}
+
+fn CoverageCounter::to_string(self : CoverageCounter) -> String {

```

Comment:
is this for debugging? if so, derive would be enough

Reply:
It's for exporting the coverage data in `@coverage.end()`, so it must be in JSON format. I don't know if deriving has such guarantees.

Reply:
Update: converted it to a custom implementation of `Debug` and derived `Show`